src/rt/backtrace/dwarf.d

```diff
@@ -0,0 +1,464 @@
+                    machine.isEpilogueBegin = false;
+                    break;
+
+                case advancePC:
+                    ulong op = readULEB128(program);
+                    // trace("advancePC ", op * lpHeader.minimumInstructionLength);
+                    machine.address += op * lpHeader.minimumInstructionLength;
+                    break;
+
+                case advanceLine:
+                    long ad = readSLEB128(program);
+                    // trace("advanceLine ", ad);
+                    machine.line += ad;
+                    break;
+
+                case setFile:
+                    uint index = cast(uint) readULEB128(program);
+                    // trace("setFile to ", index);
+                    machine.fileIndex = index;
+                    break;
+
+                case setColumn:
+                    uint col = cast(uint) readULEB128(program);
+                    // trace("setColumn ", col);
+                    machine.column = col;
+                    break;
+
+                case negateStatement:
+                    // trace("negateStatement");
+                    machine.isStatement = !machine.isStatement;
+                    break;
+
+                case setBasicBlock:
+                    // trace("setBasicBlock");
+                    machine.isBasicBlock = true;
+                    break;
+
+                case constAddPC:
+                    machine.address += (255 - lpHeader.opcodeBase) / lpHeader.lineRange * lpHeader.minimumInstructionLength;
+                    // trace("constAddPC ", "0x%x".format(machine.address));
+                    break;
+
+                case fixedAdvancePC:
+                    uint add = program.read!uint();
+                    // trace("fixedAdvancePC ", add);
+                    machine.address += add;
+                    break;
+
+                case setPrologueEnd:
+                    machine.isPrologueEnd = true;
+                    // trace("setPrologueEnd");
+                    break;
+
+                case setEpilogueBegin:
+                    machine.isEpilogueBegin = true;
+                    // trace("setEpilogueBegin");
+                    break;
+
+                case setISA:
+                    machine.isa = cast(uint) readULEB128(program);
+                    // trace("setISA ", m.isa);
+                    break;
+
+                default:
+                    // unimplemented/invalid opcode
+                    return false;
+            }
+        }
+        else
+        {
+            opcode -= lpHeader.opcodeBase;
+            auto ainc = (opcode / lpHeader.lineRange) * lpHeader.minimumInstructionLength;
+            machine.address += ainc;
+            auto linc = lpHeader.lineBase + (opcode % lpHeader.lineRange);
+            machine.line += linc;
+
+            // trace("special ", ainc, " ", linc);
+            if (!callback(machine.address, LocationInfo(machine.fileIndex, machine.line))) return true;
```

Comment:
Looks like it could be easily turned into a Range instead of using the callback scheme, but that's secondary.